File: server/src/main/java/com/example/dq/model/DataSourceRequest.java
package com.example.dq.model;

import jakarta.validation.constraints.NotBlank;

/** 数据源新增/编辑请求。password 为空表示不修改(编辑时)。 */
public record DataSourceRequest(
        @NotBlank String name,
        @NotBlank String jdbcUrl,
        String username,
        String password,
        Long rowThreshold,
        Long sizeThresholdBytes
) {
}

File: desktop/src/main/kotlin/com/example/dq/model/DataSourceRequest.kt
package com.example.dq.model

/** 数据源新增/编辑请求。password 为空表示不修改(编辑时)。 */
data class DataSourceRequest(
    val name: String?,
    val jdbcUrl: String?,
    val username: String?,
    val password: String?,
    val rowThreshold: Long?,
    val sizeThresholdBytes: Long?
)
